swift/ql/src/queries/Security/CWE-135/StringLengthConflation.qhelp

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>Using a length value from an <code>NSString</code> in a <code>String</code>, or a count from a <code>String</code> in an <code>NSString</code>, may cause unexpected behavior. This is because certain unicode sequences are represented as one character in a <code>String</code> but as a sequence of multiple characters in an <code>NSString</code>. For example, a 'thumbs up' emoji with a skin tone modifier (&#x1F44D;&#x1F3FF;) is represented as U+1F44D (&#x1F44D;) then the modifier U+1F3FF.</p>
```

Comment:
On reflection I wonder if I should use a stronger phrase than "unexpected behavior"?  It turns out that it is possible to directly cause a crash / buffer overrun using an invalid `NSRange` to operate on an `NSString` (at least in an online Swift playground), which I had not expected.

Reply:
NSString is actually an Objective C class, so I'm not terribly surprised. I'd definitely use a stronger phrase.

Reply:
Agreed

Reply:
Done.  I've gone with:
```
may cause unexpected behavior including (in some cases) buffer overwrites.
```